From a81abb11a45bc8179ed22107518052707f345a18 Mon Sep 17 00:00:00 2001
From: Tom Hochstein <tom.hochstein@nxp.com>
Date: Fri, 2 May 2025 11:11:52 -0500
Subject: [PATCH] build: Simplify toolchain integration [YOCIMX-8897]

The dependencies for the build do not go beyond the regular userspace
sysroot, so just use that.

Upstream-Status: Submitted
Signed-off-by: Tom Hochstein <tom.hochstein@nxp.com>
---
 app/Makefile  | 4 +---
 ebpf/Makefile | 9 +++------
 2 files changed, 4 insertions(+), 9 deletions(-)

diff --git a/app/Makefile b/app/Makefile
index 1414e4e..ef4d5ea 100644
--- a/app/Makefile
+++ b/app/Makefile
@@ -6,10 +6,8 @@ ARCH ?= arm64
 llc ?= llc
 clang ?= clang
 INSTALLDIR ?= ../deploy_xdp
-KERNEL_HEADERS ?= $(SDKTARGETSYSROOT)/usr
 
-C_FLAGS := -Wall \
-	-I$(KERNEL_HEADERS)/include 
+C_FLAGS := -Wall
 C_FLAGS += -I../include/	
 L_FLAGS := -lz -lelf -lbpf
 
diff --git a/ebpf/Makefile b/ebpf/Makefile
index 2344c08..cb56904 100644
--- a/ebpf/Makefile
+++ b/ebpf/Makefile
@@ -4,16 +4,13 @@
 CC ?= gcc
 ARCH ?= arm64
 llc ?= llc
-clang ?= clang
+clang ?= clang --sysroot=$(SDKTARGETSYSROOT)
 INSTALLDIR ?= ../deploy_xdp
-KERNEL_HEADERS ?= $(SDKTARGETSYSROOT)/usr
 
-CL_FLAGS := -g -O2 -emit-llvm -target bpf -D__aarch64__ -D__KERNEL__ -D__BPF__ -Wall \
-	-I$(KERNEL_HEADERS)/include
+CL_FLAGS := -g -O2 -emit-llvm -target bpf -D__aarch64__ -D__KERNEL__ -D__BPF__ -Wall
 
 
-C_FLAGS := -Wall \
-	-I$(KERNEL_HEADERS)/include 
+C_FLAGS := -Wall
 CL_FLAGS += -I../include
 C_FLAGS += -I../include/	
 L_FLAGS := -lzstd -lz -lelf -lbpf
-- 
2.34.1

